src/uct/cuda/base/cuda_md.c

```diff
@@ -94,24 +91,44 @@ uct_cuda_base_get_base_addr_alloc_length(uct_cuda_copy_md_t *md,
         if (cu_err != CUDA_SUCCESS) {
             cuGetErrorString(cu_err, &cu_err_str);
             ucs_error("cuDeviceTotalMem error: %s", cu_err_str);
-            return UCS_ERR_INVALID_ADDR;
+            goto err;
         }
     }
 
     ucs_spin_unlock(&uct_cuda_base_lock);
 
+    return total_bytes[cuda_device];
+
+err:
+    return 1; /* return 1 byte to avoid division by zero */
+}
+
+static ucs_status_t
+uct_cuda_base_get_base_addr_alloc_length(uct_cuda_copy_md_t *md,
+                                         CUdevice cuda_device,
+                                         const void *address,
+                                         size_t length,
+                                         void **base_address,
+                                         size_t *alloc_length)
+{
+    size_t total_bytes = uct_cuda_base_get_total_device_mem(cuda_device);
+    CUresult cu_err;
+    const char *cu_err_str;
+    CUdeviceptr base_addr;
+    size_t alloc_len;
+    double ratio;
+
     cu_err = cuMemGetAddressRange(&base_addr, &alloc_len, (CUdeviceptr)address);
     if (cu_err != CUDA_SUCCESS) {
         cuGetErrorString(cu_err, &cu_err_str);
         ucs_error("cuMemGetAddressRange(%p) error: %s", address, cu_err_str);
         return UCS_ERR_INVALID_ADDR;
     }
 
-    ratio = (alloc_len / total_bytes[cuda_device]);
+    ratio = (alloc_len / total_bytes);
```

Comment:
won't it always be 0??

Reply:
total_bytes = (device == T4 || error) ? 1: actual_total_bytes(between 1GB, 64GB);
ratio = 0 iff alloc_len = 0 (never true) or total_bytes = inf (also never true). Am I missing something?

Reply:
if device is not T4:
alloc_len = 1GB (for example)
total_bytes=2GB (for example)
since (alloc_len < total_bytes) then alloc_len/total_bytes is 0 (integer division)
then ratio=0
but actually it should be 0.5

Reply:
>  since (alloc_len < total_bytes) then alloc_len/total_bytes is 0 (integer division)

My bad. I forgot that alloc_len is not floating point. I'll change the type here.